doc/news/_preparation_next_release.md

```diff
@@ -80,6 +80,10 @@ The following section lists news about the [modules](https://www.libelektra.org/
 
 - We now treat relative paths as relative to `KDB_DB_SPEC` instead of the current working directory. _(Klemens Böswirth)_
 
+### Spec
+
+- There is now the config key `missing/log` that allows logging of all missing `require`d keys. _(Klemens Böswirth)_
```

Comment:
Shouldn't we fail if a required key is missing? Why do we need the log as this should already be part of the error message?

Reply:
1. `spec` can be configured to ignore some or all errors. I don't really know why, but it's always been the case.
2. This is needed for the case where a program using the codegen API is called in "help mode", i.e. `app -h` or `app --help`. In such a case, we want to ignore missing required keys and simply print the help message.
How the `missing/log` array helps detect this case, is documented in the relevant code:
https://github.com/ElektraInitiative/libelektra/blob/ea74e58203edeaf9175d65f091753d286887da45/src/libs/highlevel/elektra.c#L143-L158
There could technically be other problems from the `spec` plugin as well, but since `gopts` runs first and was successful, there can't be any errors that would break the help message.

Reply:
> spec can be configured to ignore some or all errors. I don't really know why, but it's always been the case.

This is one of spec's anti-features. Thank you for documenting it but I would be in favor of removing these anti-features. If validation fails, the plugin should return an error.

> This is needed for the case where a program using the codegen API is called in "help mode"

Ok. But wouldn't it be better to directly ask for a meta-key which says that we are in the help mode? We want the help to be printed no matter what errors happened in `kdbOpen` or `kdbGet`.

Reply:
> We usually only add warnings during kdbGet()

There are many cases, where `kdbGet()` should return an error. The best example would be a broken storage file. The storage plugin should return an error and `kdbGet()` should abort.

For the `spec` plugin, it probably makes sense to have to option of switching between warnings and errors. In `kdb` (and other configuration tools) warnings would be preferred, as a `kdbGet()` is required before a `kdbSet()` could fix the problems. In applications on the other hand errors are the better solution. They are easier to detect, since `kdbGet()` uses a different return code for errors. The handling of actual errors is also easier in the high-level API (see #2952).

> Can you make a decision of what you wrote here?

Will do.

Reply:
> There are many cases, where kdbGet() should return an error.

I agree. I made a different proposal: #2955

> For the spec plugin, it probably makes sense to have to option of switching between warnings and errors.

I hope this will not be needed once we have #2955 implemented

>   Will do.

:sparkling_heart: